templates/common/baremetal/units/nodeip-configuration.service.yaml

```diff
@@ -13,14 +13,20 @@ contents: |
   [Service]
   # Need oneshot to delay kubelet
   Type=oneshot
-  ExecStart=/usr/bin/podman run --rm \
+  # Would prefer to do Restart=on-failure instead of this bash retry loop, but
+  # the version of systemd we have right now doesn't support it. It should be
+  # available in systemd v244 and higher.
+  ExecStart=/bin/bash -c " \
+    while :; do \
+    /usr/bin/podman run --rm \
     --authfile /var/lib/kubelet/config.json \
     --net=host \
     --volume /etc/systemd/system:/etc/systemd/system:z \
     {{ .Images.baremetalRuntimeCfgImage }} \
     node-ip \
     set --retry-on-failure \
-    {{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP }}
+    {{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP }} && break; \
```

Comment:
Can you add a pause in here, just in case this is a rapid failure? Maybe something like:

```bash
until <command>
do
  sleep 5
done
```

Reply:
Yep, good idea. Done.